src/etc/inc/unbound.inc

```diff
@@ -767,6 +787,25 @@ function unbound_generate_zone_data($domain, $hosts, &$added_ptr, $zone_type = "
 	return $zone_data;
 }
 
+function unbound_generate_respip_data($nat_entry) {
+
+	if (is_ipaddrv4($nat_entry['external'])) {
+		$type = 'A';
+		$external_addr = "{$nat_entry['external']}/32";
+	} else if (is_ipaddrv6($nat_entry['external'])) {
+		$type = 'AAAA';
+		$external_addr = "{$nat_entry['external']}/128";
+	} else {
+		return "";
+	}
+	$redirect_addr = $nat_entry['source']['address'];
```

Comment:
This looks like it would be OK for the basic case when a 1:1 NAT entry is for an IP address set to "Single Host" but when about when it's for a subnet?

If the internal IP address type is set to Network and given a subnet mask, it makes a whole subnet's worth of 1:1 NAT in a single entry (e.g. `x.x.x.0/24 <-> y.y.y.0/24`). Similar case for the interface network macros as they are also networks, not addresses

If you want to restrict this to acting on single addresses only, that would make sense, but it would need to test the internal address type to make sure it's only what you want to allow (single host and maybe the "<interface> Address" macros).

If you do restrict the behavior, be sure to note that in the option help as well.

Reply:
I did a significant amount of testing including for large IP ranges and this should work properly.